applications/ShapeOptimizationApplication/custom_utilities/damping/damping_utilities.h

```diff
@@ -155,63 +151,63 @@ class DampingUtilities
+            const auto p_damping_function = CreateDampingFunction(dampingFunctionType, dampingRadius );
 
             // Loop over all nodes in specified damping sub-model part
-            for(auto& node_i : dampingRegion.Nodes())
-            {
-                ModelPart::NodeType& currentDampingNode = node_i;
+            block_for_each(dampingRegion.Nodes(), [&](const ModelPart::NodeType& rNode) {
                 NodeVector neighbor_nodes( mMaxNeighborNodes );
                 DoubleVector resulting_squared_distances( mMaxNeighborNodes,0.0 );
-                unsigned int number_of_neighbors = mpSearchTree->SearchInRadius( currentDampingNode,
+                const unsigned int number_of_neighbors = mpSearchTree->SearchInRadius( rNode,
                                                                                  dampingRadius,
                                                                                  neighbor_nodes.begin(),
                                                                                  resulting_squared_distances.begin(),
                                                                                  mMaxNeighborNodes );
 
-                ThrowWarningIfNodeNeighborsExceedLimit( currentDampingNode, number_of_neighbors );
+                ThrowWarningIfNodeNeighborsExceedLimit( rNode, number_of_neighbors );
 
                 // Loop over all nodes in radius (including node on damping region itself)
                 for(unsigned int j_itr = 0 ; j_itr<number_of_neighbors ; j_itr++)
                 {
                     ModelPart::NodeType& neighbor_node = *neighbor_nodes[j_itr];
-                    const double dampingFactor = 1-p_damping_function->ComputeWeight( currentDampingNode.Coordinates(), neighbor_node.Coordinates());
+                    const double dampingFactor = 1.0 - p_damping_function->ComputeWeight( rNode.Coordinates(), neighbor_node.Coordinates());
 
                     // For every specified damping direction we check if new damping factor is smaller than the assigned one for current node.
                     // In case yes, we overwrite the value. This ensures that the damping factor of a node is computed by its closest distance to the damping region
-                    auto& damping_factor_variable = neighbor_node.GetValue(DAMPING_FACTOR);
+                    array_3d& damping_factor_variable = neighbor_node.GetValue(DAMPING_FACTOR);
+                    neighbor_node.SetLock();
                     if(dampX && dampingFactor < damping_factor_variable[0])
                         damping_factor_variable[0] = dampingFactor;
                     if(dampY && dampingFactor < damping_factor_variable[1])
                         damping_factor_variable[1] = dampingFactor;
                     if(dampZ && dampingFactor < damping_factor_variable[2])
                         damping_factor_variable[2] = dampingFactor;
+                    neighbor_node.UnSetLock();
```

Comment:
@philbucher is this the proposed way to make this check and assignment thread-safe?

Reply:
since you loop over the node (and not e.g. over conditions) this access is inherently threadsafe as each node has its own `DAMPING_FACTOR` (if I understood correctly)
I.e. you don't need the locking

Reply:
Hm, i am looping here over the neighbors in an inner loop. So these operations are not thread safe, because the same node can be neighbor of several other nodes in the parallel loop. My question was more about if the `SetLock()/UnSetLock()/ are the right functions to use or if there is anything more tailored to finding the minimal value.

Reply:
ah now I understand
yes then it is correct as is